synapseclient/models/factory_operations.py

```diff
@@ -0,0 +1,1209 @@
+            version_number=version_number,
+            activity_options=activity_options,
+            table_options=table_options,
+            synapse_client=synapse_client,
+        )
+
+    elif entity_type == concrete_types.DATASET_COLLECTION_ENTITY:
+        return await _handle_table_like_entity(
+            entity_class=DatasetCollection,
+            synapse_id=synapse_id,
+            version_number=version_number,
+            activity_options=activity_options,
+            table_options=table_options,
+            synapse_client=synapse_client,
+        )
+
+    elif entity_type == concrete_types.ENTITY_VIEW:
+        return await _handle_table_like_entity(
+            entity_class=EntityView,
+            synapse_id=synapse_id,
+            version_number=version_number,
+            activity_options=activity_options,
+            table_options=table_options,
+            synapse_client=synapse_client,
+        )
+
+    elif entity_type == concrete_types.MATERIALIZED_VIEW:
+        return await _handle_table_like_entity(
+            entity_class=MaterializedView,
+            synapse_id=synapse_id,
+            version_number=version_number,
+            activity_options=activity_options,
+            table_options=table_options,
+            synapse_client=synapse_client,
+        )
+
+    elif entity_type == concrete_types.SUBMISSION_VIEW:
+        return await _handle_table_like_entity(
+            entity_class=SubmissionView,
+            synapse_id=synapse_id,
+            version_number=version_number,
+            activity_options=activity_options,
+            table_options=table_options,
+            synapse_client=synapse_client,
+        )
+
+    elif entity_type == concrete_types.VIRTUAL_TABLE:
+        return await _handle_table_like_entity(
+            entity_class=VirtualTable,
+            synapse_id=synapse_id,
+            version_number=version_number,
+            activity_options=activity_options,
+            table_options=table_options,
+            synapse_client=synapse_client,
+        )
+
+    else:
+        from synapseclient import Synapse
+
+        client = Synapse.get_client(synapse_client=synapse_client)
+        client.logger.warning(
+            "Unknown entity type: %s. Falling back to returning %s as a dictionary bundle matching "
+            "https://rest-docs.synapse.org/rest/org/sagebionetworks/repo/model/entitybundle/v2/EntityBundle.html",
+            entity_type,
+            synapse_id,
+        )
+
+        # This allows the function to handle new entity types that may be added in the future
+        if version_number is not None:
+            return await get_entity_id_version_bundle2(
+                entity_id=synapse_id,
+                version=version_number,
+                synapse_client=synapse_client,
+            )
+        else:
+            return await get_entity_id_bundle2(
+                entity_id=synapse_id, synapse_client=synapse_client
+            )
```

Comment:
Fallback behavior so when more entity types get added to Synapse this function will still return something that folks can use.